src/workloads/execution/ShardedLookup.yml

```diff
@@ -142,6 +142,40 @@ Actors:
             }}
           ]
         cursor: {batchSize: *NumDocs}
+    - OperationMetricsName: LookupWithCachedPrefixShardedToSharded
+      OperationName: RunCommand
+      OperationCommand:
+        aggregate: Collection0
+        pipeline:
+          [{
+            $lookup: {
+              from: Collection1,
+              let: {localInt: "$int"},
+              pipeline: [
+                {$group: {_id: {$sum: ["$int", 1]}}},
+                {$match: {$expr: {$eq: ["$_id", "$$localInt"]}}}
+              ],
```

Comment:
Hmm I don't think this pipeline will be able to use the cache even after [SERVER-57168](https://jira.mongodb.org/browse/SERVER-57168), since the cache stage would be abandoned if sent to the shards: https://github.com/mongodb/mongo/blob/82f713b946460738ad15469c083f173ae0d1f80b/src/mongo/db/pipeline/sharded_agg_helpers.cpp#L628.

Could you add a sharded to sharded $lookup query that can cache? The pipeline will need to have a stage that forces a pipeline split (like a $group or $limit) before the $lookup. This would have the $lookup execute on a merger rather than in parallel and would (I believe) have similar performance to a similar unsharded to sharded scenario.

Reply:
Edit: Brain outage moment! All of the above doesn't apply since the cache stage is added to the foreign pipeline, _not_ the original pipeline. Thanks for bringing that up Hana!